docs/proposals/coap.md

```diff
@@ -0,0 +1,79 @@
+    ```
+    </sensors/temp>;rt="oic.r.temperature";if="sensor",
+    </sensors/light>;rt="oic.r.light.brightness";if="sensor"
+    ```
+
+    The example is stating that the device has 2 REST resources, `/sensors/temp` and `/sensors/light` which are of type `oic.r.temperature` and `oic.r.light.brightness` respectively. `rt` values are defined in [IANA](https://www.iana.org/assignments/core-parameters/core-parameters.xhtml#rt-link-target-att-value) to have some standardization, although vendor-specific values can be used. Then `if` means that the interface description of the resource is of type `sensor`. Currently, only resources with interface `sensor` are supported by the discovery handler.
+3. For each device, the discovery handler returns a result which will have the following properties:
+    ```
+    COAP_IP:                 192.168.1.126
+    COAP_RESOURCE_TYPES:     oic.r.temperature,oic.r.light.brightness
+    oic.r.light.brightness:  /sensors/light
+    oic.r.temperature:       /sensors/temp
+    ```
+    `COAP_IP` and `COAP_RESOURCE_TYPES` are static and available for each Instance. `oic.r.light.brightness` and `oic.r.temperature` are dynamic, based on the discovered resources. By doing so, the cluster can look for a resource (e.g. temperature measurements) by searching for Instances that support the `oic.r.temperature` resource. 
+4. An Akri Broker is provisioned for each Instance. The Broker has the following environment variables based on the previous properties and can be reached via its associated service:
+
+  ```
+  COAP_IP=192.168.1.126
+  oic.r.temperature=/sensors/temp
+  oic.r.light.brightness=/sensors/light
+  COAP_RESOURCE_TYPES=oic.r.temperature,oic.r.light.brightness
+  ```
+
+  The Broker acts as an HTTP-to-CoAP Proxy. It translates RESTful HTTP requests into RESTful CoAP requests and vice versa for the response. "Cross-Protocol Proxying between CoAP and HTTP" is defined in section 10 of RFC 7252. Currently, the Broker forwards only GET requests.
+  
+  The Broker is also in an excellent position to cache CoAP responses. "Unlike HTTP, the cacheability of CoAP responses does not depend on the request method, but it depends on the Response Code." Currently, the Broker only caches CoAP responses with status code equal to `2.05 Content`, which is returned if the device is not reachable during the connection.
+
+## Outstanding Questions
+
+- Is there a better way to store the discovered resources as Configuration in the cluster?
+
+The current implementation would need a controller to accept queries about available resources and return the name of the Broker's service which can communicate to the device. The device is listed as a generic `akri.sh/coap-021dd7` resource on the node, which is too generic to be useful by any application. A better label would be `akri.sh/oic.r.temperature-021dd7`, the discovered resource. This would allow using the K8s controller for scheduling pods that need the resource.
```

Comment:
That is definitely an idea but the number of resource types is quite huge. Besides, as cluster administrator, who deploys the configurations, you may not know them in advance. Imagine you tasked to set up the clusters of a company with several geographical locations. It also means that you need two separate configurations, thus duplicate discovery handlers, for a same device that exposes both temperature and brightness as resources.

With that being said, I like the idea. It should also already be supported by using the `queryFilter` discoveryDetail like this:

```yaml
queryFilter:
  name: "rt" // resource type
  value: "oic.r.temperature"
```

It's part of the CoAP standard that the devices _can_ supported filtering by using a query filter. Even though I've implemented it, I never actually thought about using it for my use case 😄  It remains a workaround local deployments though, IMO.

Reply:
what do you mean by ", thus duplicate discovery handlers"? There would still only be one coap DH running on the node but it would be called twice, once for each config